### PR TITLE
Convert query control dropdown button to React

### PR DESCRIPTION
**Pull request by openjck**
This conversion needs to happen before we can easily modify it with the
extension API.

## What type of PR is this? (check all applicable)

Refactor

### CODE REVIEW

**Comment by openjck** [comment]
cc @washort

**Comment by ranbena** [comment]
@openjck wdym "extension API"? What does your follow up do?

**Comment by openjck** [comment]
Thanks for the tip about ant. I'll do that. It might take some time as I need to hop on another project for the next few days.

> @openjck wdym "extension API"? What does your follow up do?

That may not be the correct term. We use the [redash-stmo](https://github.com/mozilla/redash-stmo/) extensions in our instance of Redash.

**Comment by openjck** [comment]
It also may be the case that extensions are only supported in [our fork](https://github.com/mozilla/redash/). @washort should know the answer to that.

**Comment by openjck** [comment]
Updated to use Ant. The buttons are working for me, but let me know if your have any issues.

File: client/app/components/QueryControlDropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        </Menu.Item>
+      )}
+      {!props.query.isNew() && (
+        <Menu.Item>
+          <a onClick={() => props.showEmbedDialog(props.query, props.selectedTab)} data-test="ShowEmbedDialogButton">
+            <Icon type="share-alt" /> Embed elsewhere
+          </a>
+        </Menu.Item>
+      )}
+      {(props.queryExecuting || !props.queryResultData) && (
```
Comment (ranbena) [inline client/app/components/QueryControlDropdown.jsx]: This doesn't show up in the menu...
Cause this condition should be for a "disabled" state.

File: client/app/components/QueryControlDropdown.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            <Icon type="file-excel" /> Download as Excel File
+          </a>
+        </Menu.Item>
+      )}
+    </Menu>
+  );
+
+  return (
+    <Dropdown trigger={['click']} overlay={menu}>
+      <Button><Icon type="ellipsis" rotate={90} /></Button>
```
Comment (ranbena) [inline client/app/components/QueryControlDropdown.jsx]: The 15px side padding of the button, make it look too wide:

Screen Shot 2019-04-20 at 8 20 07


This should make it a square w/o extra markup:
```less
// ant.less
.@{btn-prefix-cls} .@{icon-prefix-cls}-ellipsis {
  margin: 0 -7px;
}
```
Screen Shot 2019-04-20 at 8 19 53

File: client/app/components/QueryControlDropdown.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (openjck) [reply client/app/components/QueryControlDropdown.jsx]: Whoops. Reversed the logic here. Thanks!

File: client/app/pages/queries/query.html [hunk 3]
```diff
@@ -255,36 +255,21 @@ <h3>
       <div class="bottom-controller-container" ng-if="showDataset">
         <div class="bottom-controller">
 
-          <button class="m-r-5 btn btn-default btn-edit-visualisation" ng-click="openVisualizationEditor(selectedTab)" ng-if="!query.isNew() && canEdit"><span class="fa fa-edit"></span><span class="hidden-xs hidden-s hidden-m"> Edit Visualization</span></button>
+          <edit-visualization-button
```
[the remark this answers is not in the document]
Comment (openjck) [reply client/app/pages/queries/query.html]: It looks like [react2angular doesn't pass children to the React component](https://github.com/coatue-oss/react2angular/issues/12) so I'm not sure if this is possible. Please let me know if I'm mistaken. :smiley:

File: client/app/components/QueryControlDropdown.jsx (hunk 2 above)
Comment (openjck) [reply client/app/components/QueryControlDropdown.jsx]: Good call.

**Comment by openjck** [comment]
Good tips. Updated. :smiley:

**Comment by openjck** [comment]
That cypress test may still be failing. I'm having trouble running them locally; `npm run cypress` seems to get stuck at `rm -rf ./client/dist` forever.

File: client/app/pages/queries/query.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (ranbena) [reply client/app/pages/queries/query.html]: No that's definitely true 👍

**Comment by ranbena** [comment]
The download links aren't workin'...
That's cause their Angular versions have the `query-result-link` directive which adds `onclick` logic (`queries/query-results-link.js`), so make sure to port that as well.

Regarding the failed Cypress tests - I'll send you a fix when this PR is ready.

**Comment by openjck** [comment]
Updated! :smiley:

I'm still having a strange issue where Docker gets stuck at `rm -rf ./client/dist/` when running tests locally, so unfortunately I'm not able to do that now.

**Comment by openjck** [comment]
Percy seems great, but I'm not seeing the changes that it's reporting.

**Comment by openjck** [comment]
Updated :smiley:

**Comment by ranbena** [comment]
Sent you some code changes https://github.com/openjck/redash/pull/1

**Comment by openjck** [comment]
Merged your PR. Thanks!

**Comment by openjck** [comment]
Rebased :smiley:

**Pull request merged**